synapseclient/models/project.py

```diff
@@ -286,15 +384,119 @@ async def delete(self, synapse_client: Optional[Synapse] = None) -> None:
 
         Returns:
             None
+
+        Example: Using this method
+            Delete the project from Synapse using ID
+
+                await Project(id="syn123").delete()
+
+            Delete the project from Synapse using Name
+
+                await Project(name="my_project").delete()
+
+        Raises:
+            ValueError: If the project ID or Name is not set.
+            SynapseNotFoundError: If the project is not found in Synapse.
         """
+        entity_id = await get_id(entity=self, synapse_client=synapse_client)
+
         loop = asyncio.get_event_loop()
         current_context = context.get_current()
         await loop.run_in_executor(
             None,
             lambda: run_and_attach_otel_context(
                 lambda: Synapse.get_client(synapse_client=synapse_client).delete(
-                    obj=self.id,
+                    obj=entity_id,
                 ),
                 current_context,
             ),
         )
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Project_Copy: {self.id}"
+    )
+    async def copy(
+        self,
+        destination_id: str,
+        copy_annotations: bool = True,
+        copy_wiki: bool = True,
+        exclude_types: Optional[List[str]] = None,
+        file_update_existing: bool = False,
```

Comment:
Same comment as above, now that this is on a project level, there's other entities that can be updated beyond files